src/routes/ProtectedDashboardRoute/ProtectedDashboardRoute.tsx

```diff
@@ -0,0 +1,53 @@
+import { useAppDispatch, useAppSelector } from "@/app/hooks"
+import { logoutUser } from "@/features/auth/authSlice"
+import getNewAccessToken from "@/libs/axios/getNewAccessToken"
+import {
+  getAccessTokenFromLocalStorage,
+  getRefreshTokenFromLocalStorage,
+} from "@/utils/localStorage"
+import { jwtDecode } from "jwt-decode"
+import { ReactElement, useEffect } from "react"
+import { Navigate, useLocation } from "react-router-dom"
+
+interface ProtectedDashboardRouteProps {
+  children: ReactElement
+}
+
+const ProtectedDashboardRoute = ({
+  children,
+}: ProtectedDashboardRouteProps) => {
+  const { isAuthenticated } = useAppSelector((store) => store.auth)
+  const dispatch = useAppDispatch()
+  const location = useLocation()
+  const accessToken = getAccessTokenFromLocalStorage()
+  const refreshToken = getRefreshTokenFromLocalStorage()
+
+  const checkAuth = () => {
+    if (accessToken && refreshToken) {
+      const currentDate = new Date().getTime()
+      const decodedAccessToken = jwtDecode(accessToken)
+      const decodedRefreshToken = jwtDecode(refreshToken)
+      if (decodedRefreshToken.exp! < currentDate / 1000) {
```

Comment:
Dont assure typescript that the value will be there by adding exclamation mark.

Just add this into the condition

  decodedRefreshToken.exp &&
        decodedRefreshToken.exp < currentDate / 1000